parsing/pprintast.ml

```diff
@@ -326,6 +326,9 @@ and core_type1 ctxt f x =
              | _ -> list ~first:"(" ~last:")@;" (core_type ctxt) ~sep:",@;" f l)
           l longident_loc li
     | Ptyp_variant (l, closed, low) ->
+        let first_is_inherit = match l with
+          | {Parsetree.prf_desc = Rinherit _}::_ -> true
```

Comment:
The problematic case is narrower than that since:
```ocaml
type t = [ r | `B ]
```
parses just fine.
But printing the leading `|` whenever the first field is inherited seems fine.